File: 423/Тугарев Денис/lab3/lab3.cpp
#include <iostream>
#include <math.h>
using namespace std;

double func(double x) {                             //Функция корни которой нам нужны
    return log10(1+2*x) - 2 + x;
}

double find_root (double a, double b, double e) // a и b абсциссы концов хорды , e - "эпсилон", необходимая точность
{
    while (fabs (b-a) > e) 
        {
         a = b - (b - a) * func(b) / (func(b) - func(a)); // в начале у нас a стоящая слева от знака равно это новое значение а, справа, заданное изначально, то есть a слева это x(i+1), а справа это x(i-1), в начальный момент x0 b это x(i)         
         b = a - (a - b) * func(a) / (func(a) - func(b));
        }
    return b;
} 
// a и b (i-1)-ый и i-ый члены

int main () 
{   
    double e; //Эпсилон, необходимая точность
    double x; //Корень уравнения
    cout << "Write epsilon = "; 
    cin >> e;
    x = find_root (0, 100, e); // т.к. промежуток по заданию у пользователя не спрашивается, то был выбран от 0 до 100(изначальные начало и конец хорды)
    cout << "Root: " << x << endl;
    return 0;
}

File: 425/Дручков Александр/[3]/lab3.py
# -*- coding: utf-8 -*-
"""
Created on Sun May 15 14:38:01 2022

@author: ДНС
"""
import math as m

v = len("Дручков Александр Дмитриевич")%5
print('Мой вариант:', v) # вариант 3 (метод хорд)

def func(x):
    return m.log10(1+2*x)-2+x

E = float(input("Input Epsilon: "))

def hord(a, b, e):
    it = 0
    root = None
    if(func(a)*func(b)<0):
        
        while abs( b - a ) > e:
            it += 1 
            c = a-func(a)*(b-a) / ( func(b)-func(a) )
            
            if func(c)==0 or abs(func(c)) < e:
                root = c
                break
            
            elif func(a)*func(c) < 0:
                b = c
                
            elif func(b)*func(c) < 0:
                a = c
                
    if root is None:
        print('Корень не найден')
    else:
        print('Корень X:', root)
        print('Значение функции в этой точке = ', func(root),'\n')   

hord(1, 10, E)      

